### ISSUE TITLE
About pthread

### ISSUE THREAD

**Issue by mowkee** [AlexeyAB/darknet#6793]
did you build the `pthreads` library yourself? if yes could you please tell me how to do it? What version is it exactly? Is it `2.11.0`? I tried building the last version (`2.9.1`) hosted on [sourceware](https://sourceware.org/pthreads-win32/) but it failed with the message:

`error C2011: 'timespec': 'struct' type redefinition`

but downloading the version `3.0.0` from [sourceforge](https://sourceforge.net/projects/pthreads4w/) It compiled successfully with the command `nmake clean VC` (same command failed for the above.).

also as mentioned under Files section in sourceforge:

```
New Features
------------
Other than the following, this release is feature-equivalent to v2.11.0.

This release introduces a change to pthread_t and pthread_once_t that will
affect applications that link with the library.

pthread_t: remains a struct but extends the reuse counter from 32 bits to 64
bits. On 64 bit machines the overall size of the object will not increase, we
simply put 4 bytes of padding to good use reducing the risk that the counter
could wrap around in very long-running applications from small to, effectively,
zero. The 64 bit reuse counter extends risk-free run time from months
(assuming an average thread lifetime of 1ms) to centuries (assuming an
average thread lifetime of 1ns).

pthread_once_t: removes two long-obsoleted elements and reduces it's size.

```
appreciate any help or suggestion.

**Comment by mowkee** [comment]
it's 2.9.0 it's outdated and won't compile on 64bit windows. why not use 2.11.0 or 3.0.0 instead?
